network/src/counters.rs

```diff
@@ -87,6 +87,15 @@ pub static LIBRA_NETWORK_DIRECT_SEND_BYTES: Lazy<HistogramVec> = Lazy::new(|| {
     .unwrap()
 });
 
+pub static LIBRA_NETWORK_NOISE_HANDSHAKE: Lazy<IntCounterVec> = Lazy::new(|| {
+    register_int_counter_vec!(
+        "libra_network_noise_handshake",
+        "Libra network noise handshake counter",
+        &["role_type", "network_type"]
+    )
+    .unwrap()
+});
```

Comment:
i would also add a commit with the updated grafana dashboard displaying the noise handshake counter